animation_nodes/nodes/mesh/find_shortest_path.py

```diff
@@ -0,0 +1,93 @@
+import bpy
+from bpy.props import *
+from ... data_structures import Mesh, LongList
+from ... base_types import AnimationNode, VectorizedSocket
+from ... algorithms.mesh_generation.find_shortest_path import getShortestPath, getShortestTree
+
+modeItems = [
+    ("PATH", "Path", "Find path from source vertex to other vertex", 0),
+    ("TREE", "Tree", "Find paths from source vertex(ies) to other vertices", 1)
+]
+
+pathTypeItems = [
+    ("MESH", "Mesh", "Output paths as line mesh(es)", 0),
+    ("SPLINE", "Spline", "Output paths as poly splines", 1),
+    ("STROKE", "Stroke", "Output paths as gp strokes", 2)
+]
+
+class FindShortestPathNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_FindShortestPathNode"
+    bl_label = "Find Shortest Path"
+    errorHandlingType = "EXCEPTION"
+
+    mode: EnumProperty(name = "Mode Type", default = "TREE",
+        items = modeItems, update = AnimationNode.refresh)
+
+    pathType: EnumProperty(name = "Path Type", default = "MESH",
+        items = pathTypeItems, update = AnimationNode.refresh)
+
+    joinMeshes: BoolProperty(name = "Join Meshes", default = True,
+        update = AnimationNode.refresh)
+
+    useSourceList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("Mesh", "Mesh", "mesh")
+        if self.mode == "PATH":
+            self.newInput("Integer", "Source", "source")
```

Comment:
Isn't it possible to make this vectorized even in the PATH mode. In this case, it will be the shortest path to the closest source.

Reply:
Omar, I think that may confuse the user.
To find the *shortest path to the closest source*
1. We have to find a path from all sources to the target.
2. Then check the path length for all the paths.
3. Output the path which is the shortest or closest source path.

Reply:
I am not sure about the confusion that this might cause. By default, it will be a single value. So a user will not have to deal with multiple values unless they want. Do we really have to go through all those steps? I though simply setting a list of sources will make things work. If it it is not that case, then yo can ignore this for now.

Reply:
> I am not sure about the confusion that this might cause. By default, it will be a single value. So a user will not have to deal with multiple values unless they want. 

Okay.

> Do we really have to go through all those steps?

Yes.

> I though simply setting a list of sources will make things work. If it it is not that case, then you can ignore this for now.

Okay.